addons/web/static/src/model/relational_model/relational_model.js

```diff
@@ -223,6 +236,30 @@ export class RelationalModel extends Model {
         }
     }
 
+    /**
+     * If we group by default based on a property, the property might not be loaded in `fields`.
+     */
+    async _getPropertyDefinition(config, propertyFullName) {
+        // dynamically load the property and add the definition in the fields attribute
+        const result = await this.orm.call(
+            config.resModel,
+            "get_property_definition",
+            [propertyFullName],
+            { context: config.context }
+        );
+        if (!result) {
+            // the property might have been removed
+            config.groupBy = null;
+        } else {
+            result.propertyName = result.name;
+            result.name = propertyFullName; // "xxxxx" -> "property.xxxxx"
+            // needed for _applyChanges
+            result.relatedPropertyField = { fieldName: propertyFullName.split(".")[0] };
+            result.relation = result.comodel; // match name on field
+            config.fields[propertyFullName] = result;
```

Comment:
maybe this rpc should return the field directly, instead of something else from which you build a field in js.